lib/workers/global/index.ts

```diff
@@ -85,11 +85,28 @@ export async function validatePresets(config: AllConfig): Promise<void> {
   }
 }
 
+export async function resolveGlobalExtends(
+  globalExtends: string[]
+): Promise<AllConfig> {
+  try {
+    // Make a "fake" config to pass to resolveConfigPresets and resolve globalPresets
+    const config = { extends: globalExtends };
+    const resolvedConfig = await resolveConfigPresets(config);
+    return resolvedConfig;
+  } catch (err) /* istanbul ignore next */ {
+    throw new Error(CONFIG_PRESETS_INVALID);
+  }
+}
+
 export async function start(): Promise<number> {
   let config: AllConfig;
   try {
     // read global config from file, env and cli args
     config = await getGlobalConfig();
+    if (config?.globalExtends) {
+      // resolve global presets immediately
+      config = await resolveGlobalExtends(config.globalExtends);
```

Comment:
```suggestion
      config = mergeChildConfig(config, await resolveGlobalExtends(config.globalExtends));
```
 Should do it.

I think we need to write hostrules before, so we have secrets accessable?